### DISCUSSION TITLE
Node shaking issue

### DISCUSSION THREAD

**Discussion by S1996ouvik** [sceneview/sceneview#478] [Q&A]
When i render a video or 3d object shaking issue occurs. How to fix it?

**Comment by ThomasGorisse** [comment]
Please provide more details

**Comment by S1996ouvik** [comment]
While we are scanning the marker, video is not stabilized and not steady
Code used 

https://github.com/SceneView/sceneview-android/assets/93898138/b2ed1b03-541c-4378-8fab-d5ae222417dc


`videoNode = VideoNodeNew(
                            chromaKeyColor =
                            if (!videoChroma.isNullOrEmpty()) {
                                Color.parseColor(videoChroma)
                            } else {
                                null
                            },
                            materialLoader = sceneView.materialLoader,
                            keepAspectRatio = false,
                            player = mediaPlayerMainVideo.apply {
                                setDataSource(
                                    requireContext(),
                                    Uri.parse(urrl)
                                )
                                this.prepareAsync()
                            }
                        )
                       }.also { qrCodeNode ->
                            qrCodeNode.rotation = Rotation(Float3(-90f, 0f, 0f))
                        }`


Here is the POC video attached. Any idea how to fix it.


https://github.com/SceneView/sceneview-android/assets/93898138/500ee281-7735-4416-895c-c14a90f85765

**Comment by Sourav-iOS** [comment]
@ThomasGorisse I have found the same issue. Please get me out of this.

**Comment by ThomasGorisse** [comment]
Your issue (which is not) is coming from the ARCore way of handling `AugmentedImage`s anchoring.
Please ask ARCore questions on the corresponding repository.

**Comment by ThomasGorisse** [comment]
Node shaking/jitter in AR is usually caused by unstable anchor tracking. In v4.0.0-rc.1, some tips:

1. **Ensure good lighting** — ARCore needs well-lit surfaces for stable tracking
2. **Use textured surfaces** — Blank walls/floors cause poor tracking
3. **Wait for stable tracking** — Only place anchors when `frame.camera.trackingState == TrackingState.TRACKING`
4. **Use Instant Placement carefully** — It's faster but less stable than regular plane anchors
5. **Smooth transforms** — The `sceneview-core` module includes `SmoothTransform` for interpolating jittery poses

If the shaking only occurs on specific devices, it may be a hardware limitation (gyroscope/accelerometer quality).
